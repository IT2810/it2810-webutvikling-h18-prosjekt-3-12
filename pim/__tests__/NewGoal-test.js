import "react-native";
import React from "react";
import NewGoal from "../screens/NewGoalScreen";
import Goals from "../screens/GoalsScreen";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
//Snapshot testing
describe("NewGoal snapshot", () => {
  it("should render newgoal view correctly", () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<NewGoal />);
    expect(result).toMatchSnapshot();
  });

  it("should render outer most view in newgoal view correctly", () => {
    //tests a snapshot view
    const tree = renderer.create(<NewGoal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

//Unit testing
describe("Unit testing: tests functions", () => {
  const newGoalContainer = renderer.create(<NewGoal />).getInstance();

  test("creates unique keys for the asyncstorage", () => {
    const key1 = newGoalContainer.createUniqueStoreKey();
    let key2 = "";
    setTimeout(() => {
      key2 = newGoalContainer.createUniqueStoreKey();
    }, 10);
    expect(key1).not.toBe(key2);
  });

  test('formated date is on format "Oct 02, 21:22"', () => {
    const formatedDate = newGoalContainer.formatDate();

    //Month to match regex. 1 big letter, then 2 small letters
    const letters = formatedDate.substring(0, 3);
    expect(letters).toMatch(/[A-Z]{1}[a-z]{2}/);

    //Day to match regex. 2 numbers
    const day = formatedDate.substring(4, 6);
    expect(day).toMatch(/[1-9]{2}/);

    //Clock to match regex. 2 numbers, then colon, then 2 numbers
    const clock = formatedDate.substring(8, 14);
    expect(clock).toMatch(/[1-9]{2}[:]{1}[0-9]{2}/);
  });

 
});
