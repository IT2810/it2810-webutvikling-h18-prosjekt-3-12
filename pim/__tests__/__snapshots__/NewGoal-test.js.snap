// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGoal snapshot should render newgoal view correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingBottom": 15,
      "paddingTop": 15,
    }
  }
>
  <ScrollViewMock
    style={
      Object {
        "marginHorizontal": 40,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingTop": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        New goal
      </Text>
      <Icon
        allowFontScaling={false}
        name="ios-add-circle-outline"
        size={25}
        style={
          Object {
            "marginLeft": 10,
            "marginTop": 0,
          }
        }
      />
    </Component>
    <Component
      styles={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Component
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          clearButtonMode="always"
          maxLength={30}
          onChangeText={[Function]}
          placeholder="Name of goal"
          style={
            Object {
              "borderColor": "#d6d7da",
              "borderRadius": 7,
              "borderWidth": 0.5,
              "fontSize": 16,
              "height": 40,
              "paddingLeft": 10,
              "paddingRight": 10,
              "width": "100%",
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          clearButtonMode="always"
          maxLength={100}
          onChangeText={[Function]}
          placeholder="Description of goal"
          style={
            Object {
              "borderColor": "#d6d7da",
              "borderRadius": 7,
              "borderWidth": 0.5,
              "fontSize": 16,
              "height": 100,
              "paddingBottom": 60,
              "paddingLeft": 10,
              "paddingRight": 10,
              "width": "100%",
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "width": "20%",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "rgb(76, 217, 100)",
                "borderRadius": 7,
                "color": "white",
                "fontSize": 17,
                "lineHeight": 24,
                "overflow": "hidden",
                "padding": 8,
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            Add
          </Text>
        </TouchableOpacity>
      </Component>
    </Component>
  </ScrollViewMock>
</Component>
`;

exports[`NewGoal snapshot should render outer most view in newgoal view correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingBottom": 15,
      "paddingTop": 15,
    }
  }
/>
`;
